rocsp/metrics.go

```diff
@@ -0,0 +1,48 @@
+package rocsp
+
+import (
+	"github.com/go-redis/redis/v8"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricsCollector struct {
+	rdb *redis.ClusterClient
+
+	// Stats accessible from the go-redis connector:
+	// https://pkg.go.dev/github.com/go-redis/redis@v6.15.9+incompatible/internal/pool#Stats
+	hits       *prometheus.Desc
+	misses     *prometheus.Desc
+	timeouts   *prometheus.Desc
+	totalConns *prometheus.Desc
+	idleConns  *prometheus.Desc
+	staleConns *prometheus.Desc
+}
+
+// Describe is implemented with DescribeByCollect. That's possible because the
+// Collect method will always return the same metrics with the same descriptors.
+func (dbc metricsCollector) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(dbc, ch)
+}
+
+// Collect first triggers the Redis ClusterClient's PoolStats function.
+// Then it creates constant metrics for each Stats value on the fly based
+// on the returned data.
+//
+// Note that Collect could be called concurrently, so we depend on Stats()
+// to be concurrency-safe.
```

Comment:
```suggestion
// Note that Collect could be called concurrently, so we depend on PoolStats()
// to be concurrency-safe.
```

BTW this property is documented here:

https://pkg.go.dev/github.com/go-redis/redis#Client
> Client is a Redis client representing a pool of zero or more underlying connections. It's safe for concurrent use by multiple goroutines.

Reply:
Thanks!